### ISSUE TITLE
Trying to train with GPU instead of CPU 

### ISSUE THREAD

**Issue by tedi14** [MIC-DKFZ/nnUNet#2429]
Hello, I am a student working on a project trying to understand how nnUNet works with TotalSegmentator. I was wondering how am I supposed to type in a command so the training happens on the GPUs. Also I wanted to ask how can I get to see the contouring it has made after the training? I am  quite new to all this any help and guidance will be very appreciated :) I asked on the TotalSegmentator GitHub but I am still not sure so I thought it will be worth asking here.

**Comment by gaojh135** [comment]
you can try this command: 
CUDA_VISIBLE_DEVICES=0 nnuNetv2 train 800 3d fullres 4 --disable checkpointing -tr nnUNetTrainer 1epoch
if you want to use multiple GPUs for training，you can do like this: 
nnUNetv2_train DATASET_NAME_OR_ID 2d 0 [--npz] -num_gpus X

Using multiple GPUs for training ->[how_to_use_nnunet](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/how_to_use_nnunet.md)

**Comment by tedi14** [comment]
Hi,
Thanks for the reply
It tells me that CUDA_VISIBLE_DEVICES isn't a recognised command. Have I not downloaded something?

**Comment by gaojh135** [comment]
Yeah, CMD can not recognize the command. You can try running command on Git bash.

**Comment by tedi14** [comment]
I still seem to get a similar issue 
![image](https://github.com/user-attachments/assets/b5fe5c5d-d875-4aec-a45b-1f06d9d85b21)

**Comment by gaojh135** [comment]
try to set batch size = 1?

**Comment by tedi14** [comment]
It doesn't change the error still the same issue

**Comment by gaojh135** [comment]
Can you take a complete screenshot? Start by entering commands. Try using the default trainer?

**Comment by tedi14** [comment]
![image](https://github.com/user-attachments/assets/626c0cf3-00a5-44c2-b81d-e0b22ee04d23)
![image](https://github.com/user-attachments/assets/7c9b3100-5979-466e-aa3a-29a45d6c05d0)

**Comment by gaojh135** [comment]
I noticed this error: "Not enough memory resources are available to process this command." How much memory do you have available when running this command? Were you able to observe any memory spikes or instances where the memory was fully utilized?

**Comment by tedi14** [comment]
Hi sorry for the late reply there is 96GB of memory available wouldn't that be enough?

**Comment by gaojh135** [comment]
try to delete .npy files then run the command?

**Comment by gaojh135** [comment]
Have you tried running it on Ubuntu? Windows often has various issues like this, which I frequently encounter as well during execution. It might be due to compatibility problems.
Someone else has encountered a similar issue as well. https://github.com/MIC-DKFZ/nnUNet/issues/1652

**Comment by LalithShiyam** [comment]
Hi @tedi14, 

You need to check the gpu utilisation. Please either use nvidia-smi or some option available in windows to see how much GPU is being used and what is using the GPU. If its a memory lock, you need to release it and then it should work - probably after a restart ;)

Cheers,
Lalith

**Comment by haosun-cb** [comment]
For the second issue if it is not solved yet:

We encountered exactly the same problem today, on a system of 96GB RAM. It turns out to be the data uncompressing issue (or plan and preprocess issue). We are not sure about the root cause, however, it was solved after removing all the .npy (not *_seg.npy) under `nnUNet_preprocessed` folder.

The training script then redid the dataset uncompressing and it worked. So would recommend try clear up enough disk space, delete your dataset under preprocessed folder. Retry plan and preprocess with a smaller -np number.

**Comment by TaWald** [comment]
Hey @tedi14 is this issue still persisting? If yes it would be great if you could post what the issue was and close this issue.

**Comment by MattAWard** [comment]
> For the second issue if it is not solved yet:
> 
> We encountered exactly the same problem today, on a system of 96GB RAM. It turns out to be the data uncompressing issue (or plan and preprocess issue). We are not sure about the root cause, however, it was solved after removing all the .npy (not *_seg.npy) under `nnUNet_preprocessed` folder.
> 
> The training script then redid the dataset uncompressing and it worked. So would recommend try clear up enough disk space, delete your dataset under preprocessed folder. Retry plan and preprocess with a smaller -np number.

This worked for me, thanks @haosun-cb

**Comment by TaWald** [comment]
This issue seems to be resolved. 
In case this is still relevant, feel free to reopen
